src/scopeHandler.js

```diff
@@ -84,12 +95,14 @@ export function createScopeHandler(unsafeRec, safeGlobal, sloppyGlobals) {
       // todo: allow modifications when target.hasOwnProperty(prop) and it
       // is writable, assuming we've already rejected overlap (see
       // createSafeEvaluatorFactory.factory). This TypeError gets replaced with
-      // target[prop] = value
-      if (objectHasOwnProperty(target, prop)) {
+      // endowments[prop] = value
```

Comment:
Change comment to indicate it must be replaced with `Reflect.set(endowments, prop, safeGlobal)` so endowments don't leak.